container/cgroupfs/blkio.go

```diff
@@ -0,0 +1,167 @@
+// +build linux
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cgroupfs
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/intelsdi-x/snap-plugin-collector-docker/container"
+)
+
+// Blkio implements StatGetter interface
+type Blkio struct{}
+
+// GetStats reads blkio metrics from Blkio Group from blkio.*
+func (b *Blkio) GetStats(stats *container.Statistics, opts container.GetStatOpt) error {
+	path, err := opts.GetStringValue("cgroup_path")
+	if err != nil {
+		return err
+	}
+	// Try to read CFQ stats available on all CFQ enabled kernels first
+	if blkioStats, err := getBlkioStat(filepath.Join(path, "blkio.io_serviced_recursive")); err == nil && blkioStats != nil {
+		return getCFQStats(path, stats)
+	}
+	return getStats(path, stats) // Use generic stats as fallback
```

Comment:
Could you put here (or above functions  `getCFQStats` and `getStats` ) what distinguishes them - I mean why, based on on availability of `blkio.io_serviced_recursive`, one of them is called, otherwise the second one.